File: Two_Sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
       #O(N^2) solution
        for x in range(len(nums)):
            for y in range(x+1,len(nums)):
                if(x!=y):
                    if (nums[x]+nums[y]==target):
                        return [x,y]
        return None

        #O(N) solution using dictionary to store the complements of nums value
        cMap=dict()
        for x,arrNum in enumerate(nums):
            if arrNum in cMap:
                return [cMap[arrNum],x]
            cMap[target-arrNum]=x
        return None

File: Two_Sum/two_sum.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        
        // goal is O(n)

        // lets use a hashmap and map the key to the position in the array

        var hm=new Map()

        // we will iterate over the array nums and add it to the hashmap

        for (var x=0;x<nums.length;x++){
            if(hm.has(target-nums[x])){
                return [hm.get(target - nums[x]), x];
            }
            hm.set(nums[x], x); //key is int at nums[x] and value is x which is position
        }
        return null
    }
}
